### PR TITLE
BA2027.CompileUsingCurrentDialects

### PR DESCRIPTION
**Pull request by shaopeng-gh**
have another PR that contains both PE and Elf, canceling this one

### CODE REVIEW

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Native_x64_VS2013_Default.dll.sarif [hunk 1]
```diff
@@ -734,6 +734,27 @@
...
           ]
+        },
+        {
+          "ruleId": "BA2027",
+          "ruleIndex": 31,
+          "message": {
+            "id": "Warning",
+            "arguments": [
+              "Native_x64_VS2013_Default.dll",
+              "Library: c:\\Users\\mikefan\\documents\\visual studio 2013\\Projects\\Native_x64_VS2013_Default\\Native_x64_VS2013_Default\\x64\\Debug\\stdafx.obj, Language: Cxx, Version: NotSet\r\nLibrary: c:\\Users\\mikefan\\documents\\visual studio 2013\\Projects\\Native_x64_VS2013_Default\\Native_x64_VS2013_Default\\x64\\Debug\\Native_x64_VS2013_Default.obj, Language: Cxx, Version: NotSet\r\nLibrary: c:\\Users\\mikefan\\documents\\visual studio 2013\\Projects\\Native_x64_VS2013_Default\\Native_x64_VS2013_Default\\x64\\Debug\\dllmain.obj, Language: Cxx, Version: NotSet\r\nLibrary: C:\\Program Files (x86)\\Microsoft Visual Studio 12.0\\VC\\lib\\amd64\\MSVCRTD.lib, Language: Cxx, Version: NotSet\r\n"
```
Comment (eddynaka) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Native_x64_VS2013_Default.dll.sarif:745]: >c:\\Users\\ [](http://example.com/codeflow?start=407&length=11)

this is showing private information.
Let's use just the name of the module/library

File: src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString
+        {
+            Text = RuleResources.BA2027_CompileUsingCurrentDialects_Description
+        };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.BA2027_Pass),
+            nameof(RuleResources.BA2027_Warning),
+            nameof(RuleResources.BA2027_Warning_Item),
```
Comment (eddynaka) [inline src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs:39]: >BA2027_Warning_Item [](http://example.com/codeflow?start=33&length=19)

this should not appear here

File: src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    [Export(typeof(Skimmer<BinaryAnalyzerContext>)), Export(typeof(ReportingDescriptor))]
+    public class CompileUsingCurrentDialects : WindowsBinaryAndPdbSkimmerBase
+    {
+        /// <summary>
+        /// BA2027
+        /// </summary>
+        public override string Id => RuleIds.CompileUsingCurrentDialects;
+
+        /// <summary>
+        /// /std setting enables supported C and C++ language features from the specified version of the C or C++ language standard.
```
Comment (eddynaka) [inline src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs:28]: >/std [](http://example.com/codeflow?start=12&length=4)

The '/std' setting

Also, separate this entire line in two lines.
the text is too long

File: src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                        result.Details.Language,
+                        result.Type == AnalyzeResultType.NotSet ? "NotSet" : result.Version);
+                    sb.AppendLine(line);
+                }
+
+                // '{0}' is a Windows PE that wasn't compiled with current dialects.
+                // Compile using current dialects enables current standard-specific features and behavior.
+                // To resolve this problem, pass version 17 or later on the cl.exe command-line,
+                // e.g. '/std:c++17' for C++ and '/std:c17' for C 
+                // or set the corresponding 'Language Standard' property in the 'C/C++ -> General' Configuration property page.
```
Comment (eddynaka) [inline src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs:156]: > or set  [](http://example.com/codeflow?start=18&length=8)

Check the other rules to get some examples and the horizontal length that we normally use.
The text does not look right and there are lines that are too long

File: src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs (hunk 3 above)
Comment (shaopeng-gh) [reply src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs:28]: Sounds good will fix

File: src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs (hunk 2 above)
Comment (shaopeng-gh) [reply src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs:39]: removed

File: src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs (hunk 3 above)
Comment (shaopeng-gh) [reply src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs:28]: fixed

File: src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs (hunk 4 above)
Comment (shaopeng-gh) [reply src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs:156]: I looked the BA2021 how long it is and using the same as that one, align.

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Native_x64_VS2013_Default.dll.sarif (hunk 1 above)
Comment (shaopeng-gh) [reply src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Native_x64_VS2013_Default.dll.sarif:745]: fixed

**Pull request closed without merging**